### PR TITLE
Add property operator: giv

### LINKED ISSUE #128: add keyword for the dot operator to `like`
Hey, can you consider adding the keyword `like` for the object dot operator.

### dogescript
```dogescript
such color much pretty 
dogument like body like style is pretty
wow pretty

plz color '#bada55'
```

### javascript
```javascript
function color(pretty) {
  document['body']['style'] = pretty // || document.body.style = pretty
  return pretty
}

color('#bada55')
```

### PR DESCRIPTION
**Pull request by AnEmortalKid**
Fixes #128 

Adds an operator that behaves like `.` when accessing properties.

Once #191 is merged and this is merged, we might be able to use the logic in 191 to support `giv`. I've left it separate at the moment. `lvl` will also be a property accessor, but might not play into the same way the binary operators do (basically a replacement for a character)

### CODE REVIEW

File: lib/parser.js [hunk 1]
```diff
@@ -1189,7 +1231,11 @@ function parseStatement(parseContext) {
   // if there's more tokens, split them by a ' ' similar to how joinTokens would work
   if(tokens.length > 0)
   {
-    statement += ' ';
+    // peek at the next token and don't apply space since it would break
+    if(shouldSeparateTokens(tokens))
```
Comment (vpzomtrrfrt) [inline lib/parser.js]: This doesn't seem to be necessary, but it does make the output slightly nicer

File: test/spec/property-operators/giv/assignment/property-to-value/source.djs [hunk 2]
```diff
@@ -0,0 +1 @@
+document giv body is 'blah'
```
Comment (vpzomtrrfrt) [inline test/spec/property-operators/giv/assignment/property-to-value/source.djs]: Maybe we shouldn't assign strings to elements?  Trying this in Firefox actually throws a TypeError

File: lib/parser.js (hunk 1 above)
Comment (AnEmortalKid) [reply lib/parser.js]: ah, does `document . window` work? i'll double check on the console

Comment (AnEmortalKid) [reply lib/parser.js]:
```
$ node
> x = {
... a: 5
... }
{ a: 5 }
> x . a
5
>

```

I could go either way, without the separate token bit it just makes things slightly simpler

File: test/spec/property-operators/giv/assignment/property-to-value/source.djs (hunk 2 above)
Comment (AnEmortalKid) [reply test/spec/property-operators/giv/assignment/property-to-value/source.djs]: well i only set this as the test, but i can update it to an html string? idk what would be more appropriate. these only get compared to the parsed out JS and not actually ran (the js)

File: lib/parser.js (hunk 1 above)
Comment (AnEmortalKid) [reply lib/parser.js]: @vpzomtrrfrt any preference? looks like the beautyfier would actually combine those, so my vote is no-fancy thing and just let it separate them!

Comment (AnEmortalKid) [reply lib/parser.js]: Since it simplifies our stuff, i'm gonna get rid of the check !

**Pull request merged**